conda/plan.py

```diff
@@ -561,7 +560,6 @@ def add_unlink_options_for_update(actions, required_solves, index):
 
 def get_resolve_object(index, prefix):
     # instantiate resolve object
-    supplement_index_with_prefix(index, prefix, {})
```

Comment:
I don't think we need this any longer.  `get_index()` should always return exactly what we want, and then ideally it would be made immutable after it's delivered by `get_index()`.